spec.bs

```diff
@@ -245,6 +245,77 @@ Run the [=update document frozenness steps=] given <var ignore>child document</v
 
 Each {{HTMLMediaElement}} has a <dfn for="HTMLMediaElement">resume frozen flag</dfn>, which is initially set to false.
 
+Modifications to the Service Worker Standard {#serviceworker-mod}
+--------------------------------------------
+
+### Service Worker: <a href="https://w3c.github.io/ServiceWorker/#client-interface">Client Interface</a> ### {#serviceworker-client-dfn}
+
+<pre class="idl">
+    partial interface Client {
+        readonly attribute  ClientState state;
+    };
+
+    enum ClientState {
+        "active",
+        "frozen"
+    };
+</pre>
+
+A {{Client}} object has an associated <dfn id="dfn-service-worker-client-state" for="Client">state</dfn>, which is one of {{ClientState}} attribute value.
+
+<section>
+  <h4 id="service-worker-client-state">{{ServiceWorkerClient/state}}</h4>
+
+  The <dfn attribute for="ServiceWorkerClient">state</dfn> attribute *must* return the [=context object=]'s [=Client/state=].
+</section>
+
+### Service Worker: <a href="https://w3c.github.io/ServiceWorker/#clients-interface">Clients Interface</a> ### {#serviceworker-clients-dfn}
+
+<pre class="idl">
+    partial dictionary ClientQueryOptions {
+        ClientStateQuery state = "active";
+    };
+
+    enum ClientStateQuery {
+        "active",
+        "frozen",
+        "all"
+    };
+</pre>
+
+Adjust <a href="https://w3c.github.io/ServiceWorker/#clients-matchall">Match All</a> algorithm:
```

Comment:
Shouldn't this be under the "Algorithms" heading?

Reply:
Done